AdyenSession/Session+PartialPaymentDelegate.swift

```diff
@@ -15,25 +15,72 @@ extension Session: PartialPaymentDelegate {
         let request = BalanceCheckRequest(sessionId: sessionContext.identifier,
                                           sessionData: sessionContext.data,
                                           data: data)
-        apiClient.perform(request) { _ in
-            // TODO: Handle result
+        apiClient.perform(request) { [weak self] result in
+            self?.handle(result: result, completion: completion)
         }
     }
     
+    private func handle(result: Result<BalanceCheckResponse, Error>,
+                        completion: @escaping (Result<Balance, Error>) -> Void) {
+        switch result {
+        case let .success(response):
+            handle(response: response, completion: completion)
+        case let .failure(error):
+            handle(error: error, completion: completion)
+        }
+    }
+
+    private func handle(response: BalanceCheckResponse, completion: @escaping (Result<Balance, Error>) -> Void) {
+        guard let availableAmount = response.balance else {
+            let error = BalanceChecker.Error.zeroBalance
+            finish(with: error)
+            completion(.failure(error))
+            return
+        }
+        let balance = Balance(availableAmount: availableAmount, transactionLimit: response.transactionLimit)
+        completion(.success(balance))
+    }
+
+    private func handle(error: Error, completion: @escaping (Result<Balance, Error>) -> Void) {
+        finish(with: error)
+        completion(.failure(error))
```

Comment:
maybe `completion` should be called before `finish()` since technically we're not finished until completion is called 🤔